cmake/mingw32-w64-cross-travis.toolchain.cmake

```diff
@@ -0,0 +1,53 @@
+#
+# Mingw-w64 cross compiler toolchain
+#
+# - The usual CMAKE variables will point to the cross compiler
+# - HOST_EXE_LINKER, HOST_C_COMPILER, HOST_EXE_LINKER_FLAGS,
+#   HOST_C_FLAGS point to a host compiler
+#
+
+set(MINGW_TRIPLET i686-w64-mingw32)
+# For x86_64 use
+#set(MINGW_TRIPLET x86_64-w64-mingw32)
+
+# The location of your toolchain sys-root
+set(MINGW_PREFIX_PATH /opt/mingw32/${MINGW_TRIPLET}/)
+# or sometimes like this
+#set(MINGW_PREFIX_PATH /usr/${MINGW_TRIPLET}/sys-root)
```

Comment:
> Doesn't work that well for third-party because the options are not passed through externalproject_add. We would need to change all recipes to pass those two options in configure_command for cmake.

I'm sorry, I'm confused.  How does `set(FOO ...)` work all that differently than `set(FOO ... CACHE)` in this case?  The difference here is with the former, I can't override the value of FOO, but with the latter I could.  Or is there some other interaction that I'm missing here.  All that should be different is that the CACHE version opens it up to be overridden.  Otherwise, the two should be the same.